src/rust/src/x509/verify.rs

```diff
@@ -43,6 +47,125 @@ impl CryptoOps for PyCryptoOps {
     }
 }
 
+struct FixedPolicy<'a>(Policy<'a, PyCryptoOps>);
+
+/// This enum exists solely to provide heterogeneously typed ownership for `OwnedPolicy`.
+enum SubjectOwner {
+    // TODO: Switch this to `Py<PyString>` once Pyo3's `to_str()` preserves a
+    // lifetime relationship between an a `PyString` and its borrowed `&str`
+    // reference in all limited API builds. PyO3 can't currently do that in
+    // older limited API builds because it needs `PyUnicode_AsUTF8AndSize` to do
+    // so, which was only stabilized with 3.10.
```

Comment:
Doesn't need to block this, but would be good to send a PR that implements the method for Python >= 3.10.